project-docs/linter-spec.md

```diff
@@ -0,0 +1,186 @@
+
+Each element in a recipe is called an "ingredient". Ingredients are often shared across recipes. So for example we have an ingredient called `data.browser_compatibility`, which most recipes will contain. Each ingredient may have particular requirements about how it is represented in the page: for example, to satisfy the `data.browser_compatibility` ingredient a page must include the `{{Compat}}` KumaScript macro.
+
+So at a high level, compliance to a recipe means:
+
+_the items in the page must correspond to the ingredients listed in the recipe, in the correct order, and each item must satisfy the particular requirements for its ingredient_.
+
+### Optional ingredients
+
+A recipe can mark an ingredient as optional by placing a question mark after it, like `prose.description?`. This means that the page does not have to contain a representation of the ingredient, but if it does, the representation must be properly formed according to the ingredient's specification, and the representation must appear in the place specified by the recipe.
+
+Technically, whether an ingredient is optional or not is a separate attribute of the ingredient. This means that a single ingredient may be marked as optional in one recipe and mandatory in a different recipe. In practice, in the our documentation, this does not occur: there are currently no ingredients that are optional in some recipes and mandatory in others.
+
+### Page sections
+
+Many, but not all, ingredients correspond to sections of a page. A section of a page is defined as a part of the page beginning with an `<h2>` heading and ending with either another `<h2>` or the end of the page.
+
+Sections are identified by an `id` attribute on the `<h2>`. For example, the `prose.description` section starts with an element whose opening tag is like `<h2 id="Description">`. In this document we'll refer to page sections using a notation like `h2#identifier`: for example, `h2#Description`.
+
+## Recipes
+
+### JavaScript recipes
+
+We've defined the following recipes for the JS docs:
+
+- [javascript-class](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-class.yaml)
+- [javascript-namespace](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-namespace.yaml)
+- [javascript-constructor](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-constructor.yaml)
+- [javascript-method](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-method.yaml)
+- [javascript-property](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-property.yaml)
+- [javascript-error](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-error.yaml)
+- [javascript-language-feature](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-language-feature.yaml)
+
+## Recipe ingredients
+
+In total these recipes use the following ingredients:
```

Comment:
Above you only mention JS recipes but then here you list all ingredients for all recipes. 
Should we also list CSS recipes?

Reply:
Oops, that should be fixed now.